.github/workflows/ci.yaml

```
@@ -0,0 +1,44 @@
+name: CI
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: v1.17
```

We're specifying 1.16 in the [go.mod](https://github.com/Kuadrant/kcp-glbc/blob/main/go.mod#L3) at the moment. I think it'd be best to upgrade to 1.17 (and to 1.18 hopefully as soon as it lands in the coming days) but maybe meanwhile we can stick to 1.16 in the CI as I'm aware there's some small incompatibilities
```suggestion
          go-version: v1.16
```

---

Actually, looks like we need 1.17+. Likely the version in go.mod should be updated.

---

https://github.com/Kuadrant/kcp-glbc/runs/5498276682?check_suite_focus=true